config/application_space_test.go

```diff
@@ -0,0 +1,321 @@
+				Expect(resp).To(BeNil())
+
+				var clientErr *sdkerrors.ClientError
+				Expect(errors.As(err, &clientErr)).To(BeTrue(), "is client error")
+				Expect(clientErr.Unwrap()).To(causeMatcher)
+				Expect(clientErr.Message()).To(msgMatcher)
+				Expect(clientErr.Code()).To(codeMatcher)
+			},
+			Entry("nil request", nil, BeNil(), Equal("invalid nil request"), Equal(codes.InvalidArgument)),
+		)
+
+		Context("When sending a short request", func() {
+			It("should return an length error in the response", func() {
+				req := &configpb.ReadApplicationSpaceRequest{
+					Identifier: &configpb.ReadApplicationSpaceRequest_Id{Id: "gid:like"},
+				}
+				resp, err := configClient.ReadApplicationSpace(ctx, req)
+				Expect(err).To(HaveOccurred())
+				Expect(resp).To(BeNil())
+				Expect(err).To(MatchError(ContainSubstring("Id: value length must be between 22")))
+
+			})
+		})
+
+		It("ReadId", func() {
+			req := &configpb.ReadApplicationSpaceRequest{
+				Identifier: &configpb.ReadApplicationSpaceRequest_Id{Id: "gid:like-real-appspace-id"},
+			}
+
+			beResp := &configpb.ReadApplicationSpaceResponse{
+				AppSpace: &configpb.ApplicationSpace{
+					Id: "gid:like-real-appspace-id",
+				},
+			}
+
+			mockClient.EXPECT().
+				ReadApplicationSpace(
+					gomock.Any(),
+					test.WrapMatcher(test.EqualProto(req)),
+					gomock.Any(),
+				).Return(beResp, nil)
+
+			resp, err := configClient.ReadApplicationSpace(ctx, req)
+			Expect(err).To(Succeed())
+			Expect(resp).To(test.EqualProto(beResp))
+		})
+
+		It("ReadName", func() {
+			req := &configpb.ReadApplicationSpaceRequest{
+				Identifier: &configpb.ReadApplicationSpaceRequest_Name{Name: &configpb.UniqueNameIdentifier{
+					Location: "gid:like-real-customer-id",
+					Name:     "like-real-appspace-name",
+				},
+				},
+			}
+
+			beResp := &configpb.ReadApplicationSpaceResponse{
+				AppSpace: &configpb.ApplicationSpace{
+					Name: "like-real-appspace-name",
+				},
+			}
+
+			mockClient.EXPECT().
+				ReadApplicationSpace(
+					gomock.Any(),
+					test.WrapMatcher(test.EqualProto(req)),
+					gomock.Any(),
+				).Return(beResp, nil)
+
+			resp, err := configClient.ReadApplicationSpace(ctx, req)
+			Expect(err).To(Succeed())
+			Expect(resp).To(test.EqualProto(beResp))
+		})
+	})
+	Describe("AppSpaceCreate", func() {
+		DescribeTable("Request error during Create ",
+			func(req *configpb.CreateApplicationSpaceRequest, causeMatcher, msgMatcher, codeMatcher OmegaMatcher) {
+				resp, err := configClient.CreateApplicationSpace(ctx, req)
+				Expect(err).To(HaveOccurred())
+				Expect(resp).To(BeNil())
+
+				var clientErr *sdkerrors.ClientError
+				Expect(errors.As(err, &clientErr)).To(BeTrue(), "is client error")
+				Expect(clientErr.Unwrap()).To(causeMatcher)
+				Expect(clientErr.Message()).To(msgMatcher)
+				Expect(clientErr.Code()).To(codeMatcher)
+			},
+			Entry("nil request", nil, BeNil(), Equal("invalid nil request"), Equal(codes.InvalidArgument)),
+		)
```

Comment:
Again, why DescribeTable when there is just 1 Entry?

Reply:
changed